### PR TITLE
Secrets Manager, Service Account, and Documentation update

### PR DESCRIPTION
**Pull request by glasnt**
This PR does a bunch of things: 

 * Removes Berglas and incorporates Secrets Manager
 * Moves to an Service Account, with all the explicit permissions that requires
 * Documentation updates from the last two, with the `.util/parse_docs.py` scraping a valid provisioning and deployment script. 

This PR does not update the Terraform provisioning automation to suit this change. 

Other notable changes: 

 * cloudbuild/build-migrate-deploy.sh is now shorter, and is now literally a build step, a migrate step, and a deploy step (with the optional allow step at the end)
 * removed the script-automation around creating a reduced Cloud SQL user, as the automation there is flaky and requires many third-party tools. The docs show how to do it, but these sections are untagged. (Terraform also does it this way)
 * changes to settings.py to import defined secrets from secrets manager. This is also used in the superuser creation script (yay reusability)
 * 🆕 debugging debugger - `.util/parse_docs.py docs/ variables` now gives a script to run that echos all the variables back at you, so you can see if you missed something

### CODE REVIEW

**Comment by glasnt** [comment]
This PR also includes the Terraform permissions to match.

File: README.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ Unicodex uses:
  * [Google Cloud SQL](https://cloud.google.com/sql/) as the managed database, via [django-environ](https://django-environ.readthedocs.io/en/latest/)
  * [Google Cloud Storage](https://cloud.google.com/storage/) as the media storage platform, via [django-storages](https://django-storages.readthedocs.io/en/latest/)
  * [Google Cloud Build](https://cloud.google.com/cloud-build/) for build and deployment automation
- * [Google KMS](https://cloud.google.com/kms/) for secret storage, via [berglas](https://github.com/GoogleCloudPlatform/berglas)
+ * [Google Secret Manager](https://cloud.google.com/solutions/secrets-management/) for managing encrypted values
```
Comment (sethvargo) [inline README.md]:
```suggestion
 * [Google Secret Manager](https://cloud.google.com/secret-manager/) for managing encrypted values
```

File: docs/40-setup-secrets.md [hunk 2]
```diff
@@ -1,164 +1,105 @@
...
 
-For **each** `SECRET` and `VALUE`:
+For **each** `SECRET` and `VALUE` and `SERVICEACCOUNT`, we would run:
 
 ```shell,exclude
 # sample code
-berglas create ${BERGLAS_BUCKET}/$SECRET $VALUE --key ${KMS_KEY}
-berglas grant ${BERGLAS_BUCKET}/$SECRET --member serviceAccount:${SA_EMAIL}
-berglas grant ${BERGLAS_BUCKET}/$SECRET --member serviceAccount:${SA_CB_EMAIL}
+gcloud beta secrets create $SECRET
```
Comment (sethvargo) [inline docs/40-setup-secrets.md]: These commands are GA now (you can drop beta)

File: docs/40-setup-secrets.md [hunk 3]
```diff
@@ -1,164 +1,105 @@
...
+
+echo -n "$VALUE" | gcloud beta secret versions add $SECRET --data-file=-
```
Comment (sethvargo) [inline docs/40-setup-secrets.md]: Maybe explain `-n`?

Comment (glasnt) [reply docs/40-setup-secrets.md]: Good idea.

**Pull request merged**